lib/worker/src/domain/vault.rs

```diff
@@ -0,0 +1,30 @@
+use regex::Regex;
+
+#[derive(Debug)]
+pub struct Vault {
```

Comment:
still using the concept namespace and relation for a vault just to keep compatibility with data already stored in the database, when if think it's appropriate and have a better idea of the pattern a vault identifier has, we can change